module/platform/darwin/HardwareIO/tests/ConversionTest.cpp

```diff
@@ -173,42 +174,42 @@ TEST_CASE("Testing the hardware position conversions to radians", "[hardware][co
     {
         INFO("Testing the inverse position conversions");
 
-        for (size_t i = 0; i < 20; ++i) {
+        for (ssize_t i = 0; i < 20; ++i) {
 
             INFO("Testing inverse motor " << i);
 
-            for (auto& test : inverseTests[i]) {
+            for (auto& test : inverseTests[size_t(i)]) {
```

Comment:
Is this cast really needed? If so then something needs to be done to fix it so it's not.